tests/test_collect_test_data_results.py

```diff
@@ -0,0 +1,163 @@
+    results = get_modified_cosine_score_results(library_spectra, test_spectra, 100)
+    np.testing.assert_almost_equal(results,
+                                   [(0.434789196140529, 0.003861003861003861, False),
+                                    (0.4955472245596076, 0.007866273352999017, False)])
+    # Test if no error happens when only 1 or 0 library spectra within mass range
+    results = get_modified_cosine_score_results(library_spectra, test_spectra, 5.56)
+    np.testing.assert_almost_equal(results[0],
+                                   (0.0, 0.0044609665427509295, False))
+    assert results[1] is None
+
+
+def test_get_cosines_score_results(test_spectra):
+    library_spectra = load_matchms_spectrum_objects_from_file(os.path.join(
+        os.path.split(os.path.dirname(__file__))[0],
+        'tests/test_files/general_test_files/100_test_spectra.pickle'))
+    result = get_cosines_score_results(library_spectra, test_spectra, 100, 0.05, 3)
+    np.testing.assert_almost_equal(result,
+                                   [(0.434789196140529, 0.0058997050147492625, False),
+                                    (0.4955472245596076, 0.007866273352999017, False)])
+    # Test if no error happens when only 1 or 0 library spectra within mass range
+    result = get_cosines_score_results(library_spectra, test_spectra, 5.56, 0.05, 0)
+    np.testing.assert_almost_equal(result[0],
+                                   (0.0, 0.004461, False))
+    assert result[1] is None
+
+
+def test_create_optimal_results(test_spectra):
+    results = create_optimal_results(test_spectra, test_spectra)
+    assert results == [(1.0, 1.0, True), (1.0, 1.0, True)]
+
+
+def test_random_results(test_spectra):
+    results = create_random_results(test_spectra,
+                                    test_spectra)
+    assert len(results) == len(test_spectra)
+
+
+def test_generate_test_results(test_spectra,
+                               ms2library_without_spectrum_id,
+                               path_to_general_test_files,
+                               tmp_path):
+    library_spectra = load_matchms_spectrum_objects_from_file(os.path.join(
+        path_to_general_test_files,
+        '100_test_spectra.pickle'))
+    generate_test_results(ms2library_without_spectrum_id,
+                          library_spectra,
+                          test_spectra,
+                          tmp_path)
+    files_made = os.listdir(tmp_path)
+    assert set(files_made) == {'cosine_score_100_da_test_results.json', 'modified_cosine_score_100_Da_test_results.json',
+                          'ms2deepscore_test_results_100_Da.json', 'ms2deepscore_test_results_all.json',
+                          'ms2query_test_results.json', 'optimal_results.json', 'random_results.json'}
+    for file in files_made:
+        result = load_json_file(os.path.join(tmp_path, file))
+        assert isinstance(result, list)
+
+
+if __name__ == "__main__":
+    pass
```

Comment:
```suggestion
```

Reply:
Not necessary.

Reply:
I have these in since my IDE (pycharm) understands that this is a test file and adds a button next to if __name__ =="main" that runs all the tests in the file.